src/app/client/src/app/modules/shared/services/exturl-util/exturl-util.service.ts

```diff
@@ -0,0 +1,25 @@
+import { ConfigService } from './../config/config.service';
+import { ResourceService } from './../resource/resource.service';
+import { ToasterService } from './../toaster/toaster.service';
+import { Injectable } from '@angular/core';
+
+@Injectable()
+export class ExternalUrlPreviewService {
+    constructor(public configService: ConfigService, public resourceService: ResourceService,
+        public toasterService: ToasterService) { }
+    /**
+      * getRedirectUrl function is to redirect to the external url link in a new tab
+      * @param playerconfigMeta Playerconfig data to get artifacturl and contentId of the item in the player
+      * @param courseId course id of the enrolled / playing course
+      * @param userId  user id of the loggedin user
+      */
+    getRedirectUrl(playerconfigMeta: any, userId?: string, courseId?: string, batchId?: string) {
```

Comment:
why is the name getRedirectUrl but it does not have any return

Reply:
name changed to generateRedirectUrl